knox.go

```diff
@@ -526,24 +531,34 @@ func (p PrincipalMux) CanAccess(acl ACL, accessType AccessType) bool {
 
 // GetID returns the first registered ID.
 func (p PrincipalMux) GetID() string {
-	if len(p.principals) == 0 {
-		return "No Principal Found"
+	return p.defaultPrincipal.GetID()
+}
+
+// Type returns the underlying type of a principal, for logging/debugging purposes.
+func (p PrincipalMux) Type() string {
+	if len(p.allPrincipals) == 1 {
+		return p.defaultPrincipal.Type()
+	}
+
+	types := []string{}
+	for provider, principal := range p.allPrincipals {
```

Comment:
nit: I think these variable names should be swapped, right?

provider is the way authentication happened, principal is the identity?

Reply:
The map is indexed by provider name, mapping to the principal, so I believe these are correct.

Reply:
See the authentication logic in decorators.go and NewPrincipalMux for how it's built but it's basically `principals[providerName] = principal`.

Reply:
yeah I was mixed up, you're good.